lib/src/line_segment.dart

```diff
@@ -0,0 +1,87 @@
+// TODO implement https://github.com/Turfjs/turf/blob/fcdaa939c905e6cfa80cca11a0e7cbb851ad1a47/packages/turf-meta/index.js#L886
+// TODO implement https://github.com/Turfjs/turf/blob/fcdaa939c905e6cfa80cca11a0e7cbb851ad1a47/packages/turf-meta/index.js#L1001
+import 'geojson.dart';
+import 'invariant.dart';
+import 'meta.dart';
+
+// export default lineSegment;
+
+/// Creates a [FeatureCollection] of 2-vertex [LineString] segments from a
+/// [LineString] or [MultiLineString] or [Polygon] and [MultiPolygon]
+/// Returns [FeatureCollection<LineString>] 2-vertex line segments
+/// For example:
+///
+/// ```dart
+/// var polygon = Polygon.fromJson({
+///     'coordinates': [
+///       [
+///         [0, 0],
+///         [1, 1],
+///         [0, 1],
+///         [0, 0],
+///       ],
+///     ];
+/// var segments = lineSegment(polygon);
+/// //addToMap
+/// var addToMap = [polygon, segments]
+
+FeatureCollection<LineString> lineSegment(GeometryObject geoJson) {
```

Comment:
GeometryObject is the **wrong type** if you want to accept Feature and FeatureCollection!

Reply:
This is, because Feature and FeatureCollection inherit from GeoJSONObject

Reply:
![GeoJSON Polymorphism (2)](https://user-images.githubusercontent.com/10634693/159875655-4f6477ee-51c1-472d-9a49-5b26b6cbc243.png)